### ISSUE TITLE
 pull-though cache and Hashicorps AWS provider v5.71.0 problems

### ISSUE THREAD

**Issue by odise** [boring-registry/boring-registry#200]
Following problems appear in the context of pull-though cache and Hashicorps AWS provider v5.71.0 problems:

The condition in [pkg/mirror/service.go](https://github.com/boring-registry/boring-registry/pull/199/files#diff-04a2e44487721f50eabe0205d432218b472f36682738156f51dc6514bc025088L164) checks for the first version in the list which is not set for v5.71.0. The `Platform` check a few lines below does exactly that. So the condition should be removed.

Otherwise this fails:
```
curl -H "Authorization: Bearer XXXX" https://localhost/v1/mirror/registry.terraform.io/hashicorp/aws/5.70.0.json
{"errors":["core.ProviderVersions doesn't contain any platforms"]}
```

While testing this I found out, that without `BORING_REGISTRY_AUTH_STATIC_TOKEN` set the jwt token will be checked in a not so ideal way in `pkg/mirror/transport.go`. We should only cast the token if it is stored in the context. Example:

```
curl https://localhost/v1/mirror/registry.terraform.io/hashicorp/aws/5.69.0.json
{"errors":["failed to type assert to string"]}
```

Please have a look at https://github.com/boring-registry/boring-registry/pull/199 for further details.

**Comment by oliviermichaelis** [comment]
Thank you for your contribution!
The project urgently needs better unit test coverage...

**Comment by oliviermichaelis** [comment]
Released in [v0.15.4](https://github.com/boring-registry/boring-registry/releases/tag/v0.15.4). Thanks!
